test/unit/org/apache/cassandra/index/sai/cql/LuceneAnalyzerTest.java

```diff
@@ -87,9 +93,10 @@ public void testNoAnalyzerOnClusteredColumn() throws Throwable
 
         execute("INSERT INTO %s (id, val) VALUES (1, 'dog')");
 
-        assertEquals(1, execute("SELECT * FROM %s WHERE val = 'dog'").size());
+        assertThatThrownBy(() -> execute("SELECT * FROM %s WHERE val : 'dog'"))
+        .isInstanceOf(InvalidRequestException.class);;
 
-        flush();
+        // Equality still works
```

Comment:
why doesn't this require ALLOW FILTERING?

Reply:
Because it is the `NoOpAnalyzer`, which is equivalent to an identity mapping from the column's raw value to its indexed value. Therefore, the match happens in the index without searching. All SAI columns get if the index is created without specifying some kind of analyzer.

Reply:
if I create an index with NoOpAnalyzer, and then describe the table, does NoOpAnalyzer show up?  if so what is the difference between analyzer=NoOpAnalyzer and no analyzer at all?

Reply:
The only way to create an index with the `NoOpAnalyzer` is by omitting the `index_analyzer` option. It could alternatively be called the `DefaultAnalyzer`. When the `index_analyzer` option is supplied, the SAI will attempt to create a `LuceneAnalyzer` or a `NonTokenizingAnalyzer`.

The one detail I discovered today is you can create a `LuceneAnalyzer` using their `KeywordAnalyzer`, which is an identity mapper. However, the current logic is only capable of recognizing the `NoOpAnalyzer` as the identify function.

Reply:
I did a quick test describing a table. Assuming I did it correctly, the describe query just returned the creation query, which did not include any information about the analyzer in the `NoOpAnalyzer` case.

Reply:
TIL, thanks